backend/api/v2beta1/artifacts.proto

```diff
@@ -0,0 +1,125 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+option go_package = "github.com/kubeflow/pipelines/backend/api/v2beta1/go_client";
+package kubeflow.pipelines.backend.api.v2beta1;
+
+import "google/api/annotations.proto";
+import "google/api/httpbody.proto";
+import "google/rpc/status.proto";
+import "google/protobuf/timestamp.proto";
+
+service ArtifactService {
+  // Finds all artifacts within the specified namespace.
+  // Namespace field is required. In multi-user mode, the caller
+  // requires RBAC verb "list" on resources "artifacts" for the
+  // specified namespace.
+  rpc ListArtifacts(ListArtifactRequest) returns (ListArtifactResponse) {
+    option (google.api.http) = {
+      get: "/apis/v2beta1/artifacts"
+    };
+  }
+
+  // Finds a specific Artifact by ID.
+  rpc GetArtifact(GetArtifactRequest) returns (Artifact) {
+    option (google.api.http) = {
+      get: "/apis/v2beta1/artifacts/{artifact_id}"
+    };
+  }
+}
+
+message Artifact {
+  // Unique Artifact ID. Generated by MLMD.
+  string artifact_id = 1;
+  // Storage Provider to which this Artifact is located (e.g. S3, Minio, etc.).
+  string storage_provider = 3;
+  // The path location of this Artifact within the storage provider.
+  // For example an object located at s3://my-bucket/path/a/b/c will
+  // result in "path/a/b/c".
+  string storage_path = 4;
+  // The Artifact URI
+  string uri = 5;
+  // Optional Output. Specifies a signed-url that can be used to
+  // download this Artifact directly from its store.
+  string share_url = 6;
+  // The namespace associated with this Artifact. This is determined
+  // by the namespace of the parent PipelineRun that
+  // the creation of this Artifact.
+  string namespace = 7;
+  // The artifacts type (e.g. system.Dataset)
+  string artifact_type = 8;
+  // The size of the artifact in bytes.
+  // If the artifact does not exist in object store (e.g. Metrics)
+  // then this is omitted.
+  int64 artifact_size = 9;
+  // Creation time of the artifact.
+  google.protobuf.Timestamp created_at = 10;
+  // Last update time of the artifact.
+  google.protobuf.Timestamp last_updated_at = 11;
+
+  // In case any error happens retrieving a artifact field, only artifact ID
+  // and the error message is returned. Client has the flexibility of choosing
+  // how to handle the error. This is especially useful during listing call.
+  google.rpc.Status error = 12;
+}
+
+message GetArtifactRequest {
+  // Required. The ID of the artifact to be retrieved.
+  string artifact_id = 1;
+  // Optional. Set to "true" to included a signed URL with
+  // an expiry (default 15 seconds, unless configured other wise).
+  // This URL can be used to download the Artifact directly from
+  // the Artifact's storage provider. Default to false.
+  string share_url = 2;
```

Comment:
I think using ResourceView is more correct.

https://cloud.google.com/apis/design/design_patterns#resource_view

maybe a BASIC view and a DOWNLOAD view?